crates/forge_app/src/orch.rs

```diff
@@ -182,6 +181,31 @@ impl<S: AgentService> Orchestrator<S> {
         Ok(reasoning_supported)
     }
 
+    async fn set_agent_md_content(
+        &mut self,
+        mut context: Context,
+        agent: &Agent,
+    ) -> anyhow::Result<Context> {
+        let agents_content = self.services.read_agents_md().await;
```

Comment:
Initialize the orchestrator with custom rules instead of reading via the service. Its a one time operation.